tests/test_queue_overflow_strategies.py

```diff
@@ -0,0 +1,276 @@
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+IN THE SOFTWARE.
+"""
+
+from uuid import UUID
+
+import pytest
+
+from stellanow_sdk_python.message_queue.message_queue_strategy.fifo_message_queue_strategy import (
+    FifoMessageQueueStrategy,
+)
+from stellanow_sdk_python.message_queue.message_queue_strategy.lifo_message_queue_strategy import (
+    LifoMessageQueueStrategy,
+)
+from stellanow_sdk_python.message_queue.message_queue_strategy.overflow_strategy import OverflowStrategy, QueueFullError
+from stellanow_sdk_python.messages.event import StellaNowEventWrapper
+from stellanow_sdk_python.messages.message import Entity, StellaNowMessageBase, StellaNowMessageWrapper
+
+
+class SampleMessage(StellaNowMessageBase):
+    """Simple test message class."""
+
+    test_field: str
+
+
+def create_test_message(message_id: str) -> StellaNowEventWrapper:
+    """Helper to create test messages wrapped for queue."""
+    msg = SampleMessage(
+        event_name="test_event",
+        entities=[Entity(entity_type_definition_id="test", entity_id="test_entity")],
+        test_field=f"data_{message_id}",
+    )
+    wrapper = StellaNowMessageWrapper.create(msg)
+    # Override the auto-generated message_id for testing
+    wrapper.metadata.message_id = message_id
+
+    return StellaNowEventWrapper.create(
+        message=wrapper,
+        organization_id=UUID("00000000-0000-0000-0000-000000000000"),
+        project_id=UUID("00000000-0000-0000-0000-000000000000"),
+    )
+
+
+class TestFifoDropOldest:
+    """Tests for FIFO queue with DROP_OLDEST strategy."""
+
+    def test_drop_oldest_when_full(self):
+        """Test that oldest messages are dropped when queue is full."""
+        queue = FifoMessageQueueStrategy(max_size=3, overflow_strategy=OverflowStrategy.DROP_OLDEST)
+
+        # Fill queue to capacity
+        msg1 = create_test_message("msg1")
+        msg2 = create_test_message("msg2")
+        msg3 = create_test_message("msg3")
+        queue.enqueue(msg1)
+        queue.enqueue(msg2)
+        queue.enqueue(msg3)
+
+        assert queue.get_message_count() == 3
+        assert queue.get_dropped_count() == 0
+
+        # Add 4th message - should drop msg1
+        msg4 = create_test_message("msg4")
+        queue.enqueue(msg4)
+
+        assert queue.get_message_count() == 3
+        assert queue.get_dropped_count() == 1
+
+        # Dequeue and verify msg1 was dropped, msg2 is now first
```

Comment:
```suggestion
        # Dequeue and verify msg1 was taken, msg2 is now first
```

`dropped` is wrong in this context as we use `try_dequeue` to acquire message to process not to ignore it